### PR TITLE
utils: make CheckSafetyOfStartTs retry.

### PR DESCRIPTION
**Pull request by 3AceShowHand**
### What problem does this PR solve? 

This PR tries to close issue #2224 #2343 

### What is changed and how it works?

* make `CheckSafetyOfStartTs` retry 
* adjust retry option for `newStream`
* fix some typo when walking the code.

The method mentioned above may meet PD leader switch, which causes changefeed to not initialized as expected. This happens to both old and new owner implementation.

### Check List 

Tests 

 - No code

Code changes

Side effects

 - Possible performance regression

Related changes

 - Need to cherry-pick to the release branch

### Release note

No release note

### CODE REVIEW

**Comment by 3AceShowHand** [comment]
PD leader switch may take 10 ~ 30s. We have to consider about the retry options.

**Review by asddongmen — APPROVED** [review]

File: cdc/kv/client.go [hunk 1]
```diff
@@ -474,7 +474,7 @@ func (c *CDCClient) newStream(ctx context.Context, addr string, storeID uint64)
 		}
 		log.Debug("created stream to store", zap.String("addr", addr))
 		return nil
-	}, retry.WithBackoffBaseDelay(50), retry.WithMaxTries(3), retry.WithIsRetryableErr(cerror.IsRetryableError))
+	}, retry.WithBackoffBaseDelay(500), retry.WithMaxTries(60*1000), retry.WithMaxTries(8), retry.WithIsRetryableErr(cerror.IsRetryableError))
```
Comment (amyangfei) [inline cdc/kv/client.go:477]: Should `retry.WithMaxTries(60*1000)` be WithBackoffMaxDelay?

File: pkg/util/gc_service.go [hunk 2]
```diff
@@ -30,10 +33,21 @@ const (
...
+	if err := retry.Do(ctx, func() error {
+		minServiceGCTs, err = pdCli.UpdateServiceGCSafePoint(ctx, cdcChangefeedCreatingServiceGCSafePointID+changefeedID,
+			cdcChangefeedCreatingServiceGCSafePointTTL, startTs)
+		if err != nil {
+			log.Warn("update GC safepoint failed, retry later", zap.Error(err))
+		}
+		return err
+	},
+		retry.WithBackoffBaseDelay(500),
+		retry.WithBackoffMaxDelay(60*1000),
```
Comment (amyangfei) [inline pkg/util/gc_service.go:48]: Is `60*1000` used as the max delay latency of one time retry, if the max retry time is 8, the retry latency of last time is 0.5 * (1.5 ^ 8) = 25.6s, which is less than 60s, so maybe the option `retry.WithBackoffMaxDelay(60*1000)` is useless.

**Review by overvenus — COMMENTED** [review]
Rest LGTM

File: pkg/util/gc_service.go [hunk 3]
```diff
@@ -30,10 +33,20 @@ const (
...
 // and this function will update the service GC to startTs
-func CheckSafetyOfStartTs(ctx context.Context, pdCli pd.Client, changefeedID string, startTs uint64) error {
-	minServiceGCTs, err := pdCli.UpdateServiceGCSafePoint(ctx, cdcChangefeedCreatingServiceGCSafePointID+changefeedID,
-		cdcChangefeedCreatingServiceGCSafePointTTL, startTs)
-	if err != nil {
+func CheckSafetyOfStartTs(ctx context.Context, pdCli pd.Client, changefeedID string, startTs uint64) (err error) {
+	var minServiceGCTs uint64
+	// pd leader switch may happen, so just retry it.
+	if err := retry.Do(ctx, func() error {
+		minServiceGCTs, err = pdCli.UpdateServiceGCSafePoint(ctx, cdcChangefeedCreatingServiceGCSafePointID+changefeedID,
```
Comment (overvenus) [inline pkg/util/gc_service.go:40]: Could you add a test? You can use mock pd client to simulate error.

https://github.com/pingcap/ticdc/blob/596efd4b5fc05188a4937564266683aa7f2b6d8b/cdc/kv/client_mock_test.go#L28-L33

**Review by overvenus — APPROVED** [review]
LGTM

File: pkg/util/gc_service_test.go [hunk 4]
```diff
@@ -46,15 +49,42 @@ func (s *gcServiceSuite) TestCheckSafetyOfStartTs(c *check.C) {
...
+
+	s.pdCli.retryThresh = 1
+	s.pdCli.retryCount = 0
+	err = CheckSafetyOfStartTs(ctx, s.pdCli, "changefeed2", 65)
+	c.Assert(err, check.IsNil)
+
+	s.pdCli.retryThresh = 8
+	s.pdCli.retryCount = 0
+	err = CheckSafetyOfStartTs(ctx, s.pdCli, "changefeed2", 65)
+	c.Assert(err, check.Not(check.IsNil))
```
Comment (amyangfei) [inline pkg/util/gc_service_test.go:63]:
```suggestion
	c.Assert(err, check.NotNil)
```

**Review by amyangfei — APPROVED** [review]
rest lgtm

**Pull request merged**